### PR TITLE
fix: dump correct breakpad symbols on macOS

### PR DESCRIPTION
**Pull request by nornagon**
#### Description of Change
The breakpad symbols we're currently generating on macOS only include exported symbols, which makes them approximately useless for symbolication. This mimics how [Chromium](https://chromium.googlesource.com/chromium/src/+/74.0.3729.5/chrome/BUILD.gn#1331) generates breakpad symbols.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed macOS breakpad symbol files to include non-public symbols.

### CODE REVIEW

**Comment by nornagon** [comment]
(Draft state because I haven't yet integrated this into CI / release. But I've tested the GN targets and they work 👍)

**Comment by nornagon** [comment]
This is ready for review now. I've refactored how symbol generation works to avoid using the not-very-well-maintained `generate_breakpad_symbols.py` from Chromium, and to reduce the number of possible code paths. Now we just call `dump_syms` directly. Some logic was borrowed from `generate_breakpad_symbols.py` to generate the directory layout. And ninja now drives the invocation of `dump_syms`, allowing parallel execution.

**Comment by jkleinsc** [comment]
Test release builds running here:
CircleCI release build request for mas-publish successful.  Check https://circleci.com/gh/electron/electron/249751 for status.
CircleCI release build request for linux-ia32-publish successful.  Check https://circleci.com/gh/electron/electron/249753 for status.
CircleCI release build request for linux-arm64-publish successful.  Check https://circleci.com/gh/electron/electron/249755 for status.
CircleCI release build request for linux-x64-publish successful.  Check https://circleci.com/gh/electron/electron/249756 for status.
CircleCI release build request for linux-arm-publish successful.  Check https://circleci.com/gh/electron/electron/249752 for status.
CircleCI release build request for osx-publish successful.  Check https://circleci.com/gh/electron/electron/249754 for status.
AppVeyor release build request for electron-x64 successful.  Check build status at https://ci.appveyor.com/project/electron-bot/electron-x64-release/build/1.0.90
AppVeyor release build request for electron-ia32 successful.  Check build status at https://ci.appveyor.com/project/electron-bot/electron-ia32-release/build/1.0.63

File: BUILD.gn [hunk 1]
```diff
@@ -937,6 +937,76 @@ if (is_mac) {
       "@executable_path/../Frameworks",
     ]
   }
+
+  if (enable_dsyms) {
```
Comment (MarshallOfSound) [inline BUILD.gn]: This never appears to be set? Unless I'm missing a thing?

Comment (nornagon) [reply BUILD.gn]: it's set in Chromium's GN files.

**Comment by nornagon** [comment]
> The title of the PR says it is for macOS, but you are refactoring all of our symbol generation right?

It refactors all platforms, but only fixes macOS. Windows (and Linux I think? haven't checked) were already working

**Comment by deepak1556** [comment]
@nornagon Should we backport this ? Seems like this must go all the way back to 4 ?

**Comment by nornagon** [comment]
@deepak1556 probably, yeah.

File: script/release/uploaders/upload.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ def main():
   upload_electron(release, electron_zip, args)
   if get_target_arch() != 'mips64el':
     symbols_zip = os.path.join(OUT_DIR, SYMBOLS_NAME)
-    shutil.copy2(os.path.join(OUT_DIR, 'symbols.zip'), symbols_zip)
+    shutil.copy2(os.path.join(OUT_DIR, 'breakpad_symbols.zip'), symbols_zip)
```
Comment (jkleinsc) [inline script/release/uploaders/upload.py]: Where does this file actually get created?

Comment (nornagon) [reply script/release/uploaders/upload.py]: oops, i started down a path of getting gn to make the zip file too but then half-undid it. i'll revert this to `symbols.zip`, which is what `zip-symbols.py` creates.

**Comment by jkleinsc** [comment]
Release builds for testing:
CircleCI release build request for linux-arm-publish successful.  Check https://circleci.com/gh/electron/electron/250205 for status.
CircleCI release build request for linux-x64-publish successful.  Check https://circleci.com/gh/electron/electron/250207 for status.
CircleCI release build request for linux-ia32-publish successful.  Check https://circleci.com/gh/electron/electron/250206 for status.
CircleCI release build request for mas-publish successful.  Check https://circleci.com/gh/electron/electron/250204 for status.
CircleCI release build request for osx-publish successful.  Check https://circleci.com/gh/electron/electron/250209 for status.
AppVeyor release build request for electron-x64 successful.  Check build status at https://ci.appveyor.com/project/electron-bot/electron-x64-release/build/1.0.92
CircleCI release build request for linux-arm64-publish successful.  Check https://circleci.com/gh/electron/electron/250208 for status.
AppVeyor release build request for electron-ia32 successful.  Check build status at https://ci.appveyor.com/project/electron-bot/electron-ia32-release/build/1.0.65

**Comment by jkleinsc** [comment]
Release builds here:
CircleCI release build request for linux-arm64-publish successful.  Check https://circleci.com/gh/electron/electron/250481 for status.
CircleCI release build request for mas-publish successful.  Check https://circleci.com/gh/electron/electron/250477 for status.
CircleCI release build request for linux-arm-publish successful.  Check https://circleci.com/gh/electron/electron/250480 for status.
CircleCI release build request for linux-x64-publish successful.  Check https://circleci.com/gh/electron/electron/250482 for status.
CircleCI release build request for linux-ia32-publish successful.  Check https://circleci.com/gh/electron/electron/250479 for status.
CircleCI release build request for osx-publish successful.  Check https://circleci.com/gh/electron/electron/250478 for status.
AppVeyor release build request for electron-ia32 successful.  Check build status at https://ci.appveyor.com/project/electron-bot/electron-ia32-release/build/1.0.66
AppVeyor release build request for electron-x64 successful.  Check build status at https://ci.appveyor.com/project/electron-bot/electron-x64-release/build/1.0.93
 jkl

**Comment by jkleinsc** [comment]
Release builds:
CircleCI release build request for linux-ia32-publish successful.  Check https://circleci.com/gh/electron/electron/250690 for status.
CircleCI release build request for linux-x64-publish successful.  Check https://circleci.com/gh/electron/electron/250687 for status.
CircleCI release build request for mas-publish successful.  Check https://circleci.com/gh/electron/electron/250689 for status.
CircleCI release build request for linux-arm-publish successful.  Check https://circleci.com/gh/electron/electron/250691 for status.
CircleCI release build request for linux-arm64-publish successful.  Check https://circleci.com/gh/electron/electron/250688 for status.
CircleCI release build request for osx-publish successful.  Check https://circleci.com/gh/electron/electron/250692 for status.
AppVeyor release build request for electron-ia32 successful.  Check build status at https://ci.appveyor.com/project/electron-bot/electron-ia32-release/build/1.0.67
AppVeyor release build request for electron-x64 successful.  Check build status at https://ci.appveyor.com/project/electron-bot/electron-x64-release/build/1.0.94

**Comment by jkleinsc** [comment]
Release builds:
CircleCI release build request for linux-arm-publish successful.  Check https://circleci.com/gh/electron/electron/251673 for status.
CircleCI release build request for linux-x64-publish successful.  Check https://circleci.com/gh/electron/electron/251669 for status.
CircleCI release build request for linux-arm64-publish successful.  Check https://circleci.com/gh/electron/electron/251674 for status.
CircleCI release build request for linux-ia32-publish successful.  Check https://circleci.com/gh/electron/electron/251672 for status.
CircleCI release build request for osx-publish successful.  Check https://circleci.com/gh/electron/electron/251670 for status.
CircleCI release build request for mas-publish successful.  Check https://circleci.com/gh/electron/electron/251671 for status.
AppVeyor release build request for electron-ia32 successful.  Check build status at https://ci.appveyor.com/project/electron-bot/electron-ia32-release/build/1.0.69
AppVeyor release build request for electron-x64 successful.  Check build status at https://ci.appveyor.com/project/electron-bot/electron-x64-release/build/1.0.96

**Comment by jkleinsc** [comment]
Release builds:
CircleCI release build request for linux-x64-publish successful.  Check https://circleci.com/gh/electron/electron/252007 for status.
CircleCI release build request for linux-ia32-publish successful.  Check https://circleci.com/gh/electron/electron/252009 for status.
CircleCI release build request for linux-arm64-publish successful.  Check https://circleci.com/gh/electron/electron/252008 for status.
CircleCI release build request for mas-publish successful.  Check https://circleci.com/gh/electron/electron/252010 for status.
CircleCI release build request for osx-publish successful.  Check https://circleci.com/gh/electron/electron/252011 for status.
CircleCI release build request for linux-arm-publish successful.  Check https://circleci.com/gh/electron/electron/252012 for status.
AppVeyor release build request for electron-ia32 successful.  Check build status at https://ci.appveyor.com/project/electron-bot/electron-ia32-release/build/1.0.70
AppVeyor release build request for electron-x64 successful.  Check build status at https://ci.appveyor.com/project/electron-bot/electron-x64-release/build/1.0.97

**Comment by jkleinsc** [comment]
CircleCI release build request for linux-x64-publish successful.  Check https://circleci.com/gh/electron/electron/252984 for status.
CircleCI release build request for osx-publish successful.  Check https://circleci.com/gh/electron/electron/252983 for status.
CircleCI release build request for linux-arm-publish successful.  Check https://circleci.com/gh/electron/electron/252986 for status.
CircleCI release build request for linux-ia32-publish successful.  Check https://circleci.com/gh/electron/electron/252985 for status.
CircleCI release build request for linux-arm64-publish successful.  Check https://circleci.com/gh/electron/electron/252988 for status.
CircleCI release build request for mas-publish successful.  Check https://circleci.com/gh/electron/electron/252987 for status.
AppVeyor release build request for electron-x64 successful.  Check build status at https://ci.appveyor.com/project/electron-bot/electron-x64-release/build/1.0.99
AppVeyor release build request for electron-ia32 successful.  Check build status at https://ci.appveyor.com/project/electron-bot/electron-ia32-release/build/1.0.72

File: BUILD.gn [hunk 3]
```diff
@@ -1023,6 +1093,49 @@ if (is_mac) {
...
+      binary = "$root_out_dir/swiftshader/libEGL$_target_shared_library_suffix"
+      symbol_dir = "$root_out_dir/breakpad_symbols"
+      deps = [
+        "//third_party/swiftshader/src/OpenGL/libEGL:swiftshader_libEGL",
+      ]
+    }
+
+    extract_symbols("swiftshader_gles_symbols") {
+      binary =
+          "$root_out_dir/swiftshader/libGLESv2$_target_shared_library_suffix"
```
Comment (poiru) [inline BUILD.gn:1124]: @nornagon The symbols for libGLESv2.dll are not present in Sentry even though the other Electron symbols are there. See: https://sentry.io/share/issue/d07971e24b74434f9d3bbfa520b47e0a/

Do you know why that may be? Are we failing to package libGLESv2.dll.pdb on Windows?

**Pull request merged**